src/main/java/com/scottlogic/filters/AuthorFilter.java

```diff
@@ -19,20 +19,14 @@ public AuthorFilter(String criteria) {
     @Override
     public List<UserPost> filter(List<UserPost> inputList) {
         List<UserPost> filteredList = new ArrayList<>();
-        List<UserPost> nullsRemoved = new ArrayList<>();
-        List<UserPost> listToBeFiltered = new ArrayList<>();
 
         if (name == null || inputList == null) {
             return filteredList;
         }
 
-        listToBeFiltered = NullPostChecker.nullPostCheck(inputList);
-        for (UserPost userPost : listToBeFiltered) {
-            if (userPost.getAuthor() != null) {
-                nullsRemoved.add(userPost);
-            }
-        }
-        filteredList = nullsRemoved.stream().filter(post -> StringUtils.containsIgnoreCase(post.getAuthor(), name)).collect(Collectors.toList());
+        filteredList = inputList.stream().filter(Objects::nonNull)
+                .filter(post -> StringUtils.containsIgnoreCase(post.getAuthor(), name))
+                .collect(Collectors.toList());
```

Comment:
This is a lot easier to read now :) I would usually put a line break after `stream()`, so that we can read a nicely aligned list of processes, but that's no biggie.